tsup.config.ts

```diff
@@ -0,0 +1,48 @@
+import { defineConfig } from 'tsup'
+
+export default defineConfig([
+  {
+    name: 'index',
+    entry: ['./src/index.ts'],
+    outDir: './lib',
+    format: ['esm', 'cjs'],
+    legacyOutput: true,
+    sourcemap: true,
+    clean: true,
+    bundle: true,
+    splitting: false,
```

Comment:
does splitting share code between these entrypoints?  if so, we might want to enable it?

I'm not sure how these output?  are they 4 discrete bundles, or 4 modules with shared code between them? (could probably look at imports after running the build locally). 

I think we want to share most of the stuff between them outside of the interceptor specific files.

Reply:
> does splitting share code between these entrypoints? if so, we might want to enable it?

I tried to set this to true (actually that's the default setting) but I couldn't notice any difference in the build output 🤔 I tried to use the `--splitting` [flag](https://tsup.egoist.dev/#code-splitting) but that also didn't seem to make any difference. The shared code seemed to be placed inside each interceptor bundle each time.

I will do some more digging to see how this is supposed to work. 

> I'm not sure how these output? are they 4 discrete bundles, or 4 modules with shared code between them? (could probably look at imports after running the build locally).

Right now it seems they are 4 discrete bundles. This is not what we want obviously, I need to tinker more with it.